mermithid/processors/TritiumSpectrum/FakeDataGenerator.py

```diff
@@ -0,0 +1,253 @@
+        if lineshape=='gaussian':
+            max_energy = nstdevs*params[0]
+            min_energy = -1000
+        elif lineshape=='simplified_scattering' or lineshape=='simplified' or lineshape=='detailed_scattering' or lineshape=='detailed':
+            max_energy = nstdevs/FWHM_convert*params[0]
+            min_energy = -1000
+
+        Kmax_eff = Kmax+max_energy #Maximum energy for data is slightly above Kmax>Q-m
+        Kmin_eff = Kmin+min_energy #Minimum is slightly below Kmin<Q-m
+
+        step_size = (Kmax_eff-Kmin_eff)/float(nsteps)
+        logger.info('Stepsize is {} eV'.format(step_size))
+
+        #Options of kinetic energies to be sampled
+        Koptions = np.arange(Kmin_eff, Kmax_eff, step_size)
+
+        if efficiency_dict is not None:
+            logger.info('Evaluating efficiencies')
+            efficiency, efficiency_error = efficiency_from_interpolation(Koptions, efficiency_dict, B_field)
+        else:
+            efficiency, efficiency_error = 1, 0
+
+        #Create array of sampled kinetic energies.
+        logger.info('Going to calculate rates')
+        time0 = time.time()
+
+        if array_method == True:
+            ratesS = convolved_spectral_rate_arrays(Koptions, Q_mean, mass, Kmin, lineshape, params, min_energy, max_energy)
+        else:
+            ratesS = [convolved_spectral_rate(K, Q_mean, mass, Kmin, lineshape, params, min_energy, max_energy) for K in Koptions]
+
+        # multiply rates by efficiency
+        ratesS = ratesS*efficiency
+
+        time1 = time.time()
+        logger.info('... signal rate took {} s'.format(time1-time0))
+
+        # background
+        if array_method == True:
+            ratesB = convolved_bkgd_rate_arrays(Koptions, Kmin, Kmax, lineshape, params, min_energy, max_energy)
+        else:
+            ratesB = [convolved_bkgd_rate(K, Kmin, Kmax, lineshape, params, min_energy, max_energy) for K in Koptions]
+
+        time2 = time.time()
+        logger.info('... background rate took {} s'.format(time2 - time1))
+
+        if err_from_B != None:
+            dE = Koptions[1] - Koptions[0]
+            #n_dE = round(max_energy/dE) #Number of steps for the narrow gaussian for energies > 0
+            n_dE = 10*err_from_B/dE
+
+            K_lineshape =  np.arange(-n_dE*dE, n_dE*dE, dE)
+            #Generating finely spaced points on a gaussian
+            gaussian_rates = gaussian(K_lineshape, [err_from_B, 0])
+
+            ratesS = convolve(ratesS, gaussian_rates, mode='same')
+            ratesB = convolve(ratesB, gaussian_rates, mode='same')
+
+
+        ratesS[ratesS<0.] = 0.
+        ratesB[ratesB<0.] = 0.
+        rate_sumS, rate_sumB = np.sum(ratesS), np.sum(ratesB)
+        probsS = np.array(ratesS)/rate_sumS
+        probsB = np.array(ratesB)/rate_sumB
+        probs = (S*probsS + B*probsB)/(S+B)
+
+
+        logger.info('Generating data')
+        time4 = time.time()
+        #KE = np.random.choice(Koptions, np.random.poisson(S+B), p = probs)
```

Comment:
It would be nice to allow the possibility to have random number of events in the spectrum; this would allow for statistical fluctuations studies on the endpoint/neutrino mass later on. Adding an option "poisson_stats" with True or False should do the trick...

Reply:
Also it could be useful to store the number of values actually generated in the processor for unit test checking...

Reply:
added option to have random number of events.